src/app/pages/settings-technical/crypto/settings-crypto.component.ts

```diff
@@ -64,51 +64,55 @@ export class SettingsCryptoComponent implements OnInit {
   }
 
   public save(content: any) {
-    if (!content.crypto) {
-      return;
-    }
-    this.cryptoSettings.type = CryptoSettingsType.CRYPTO;
-    this.cryptoSettings.crypto.formerKey = this.cryptoSettings.crypto.key;
-    this.cryptoSettings.crypto.key = content.crypto.key;
-    this.cryptoSettings.crypto.formerKeyProperties = this.cryptoSettings.crypto.keyProperties;
-    this.cryptoSettings.crypto.keyProperties = {
-      blockCypher: content.crypto.blockCypher,
-      blockSize: content.crypto.blockSize,
-      operationMode: content.crypto.operationMode,
-    };
-    // Save
-    this.spinnerService.show();
-    this.componentService.saveCryptoSettings(this.cryptoSettings).subscribe(
-      (response) => {
-        this.spinnerService.hide();
-        if (response.status === RestResponse.SUCCESS) {
-          this.messageService.showSuccessMessage('technical_settings.crypto.update_success');
-          this.refresh();
-        } else {
-          Utils.handleError(
-            JSON.stringify(response),
-            this.messageService,
-            'technical_settings.crypto.update_error'
-          );
-        }
-      },
-      (error) => {
-        this.spinnerService.hide();
-        switch (error.status) {
-          case HTTPError.OBJECT_DOES_NOT_EXIST_ERROR:
-            this.messageService.showErrorMessage('technical_settings.crypto.setting_do_not_exist');
-            break;
-          default:
-            Utils.handleHttpError(
-              error,
-              this.router,
+    if (this.cryptoSettings.crypto.migrationToBeDone) {
```

Comment:
No, the request has to be sent to the backend and the backend will return an HTTP error code and you will display the right error message.
Check HTTPError type.